lwt/util.ml

```diff
@@ -0,0 +1,37 @@
+open Xen_api
+open Lwt
+
+let is_ssl ?allow_file:(allow_file=true) uri =
+  match Uri.scheme uri with
+  | Some "https" ->
+    true
+  | Some "http" ->
+    false
+  | Some "file" when allow_file ->
+    false
+  | x ->
+    failwith (Printf.sprintf "Unsupported URI scheme: %s" (match x with None -> "None" | Some x -> x))
+
+let sockaddr_of_uri uri =
+  let use_ssl = is_ssl ~allow_file:false uri in
+  let port =
+    match Uri.port uri with
+    | None ->
+      (if use_ssl then 443 else 80)
```

Comment:
Could use `when use_ssl -> 443`

Reply:
```ocaml
| None when use_ssl -> 443
| None -> 80
```
?
I can if this is preferred indeed.

Reply:
It's a matter of taste. I think `match` expressions are easiest to read when cases are easy to identify rather than walking in and finding out it's actually not the correct case.

Reply:
Agreed. Mixing match and if when the cases are this simple might not be the best practice indeed.